bpf/lib/ipfrag.h

```diff
@@ -0,0 +1,108 @@
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause) */
+/* Copyright Authors of Cilium */
+
+#pragma once
+
+#include <linux/ip.h>
+#include "common.h"
+
+/* IP fragment information is packed in a single __s64:
+ *
+ *  63    62...42 41     40     39...32 31...    ...0
+ * +-----+-------+------+------+-------+-------------+
+ * | err |  ...  | nol4 | frag | proto | fragment id |
+ * +-----+-------+------+------+-------+-------------+
+ *
+ * err: Sign bit, negative values encode error codes.
+ * nol4: The packet doesn't have an L4 header (i.e. not the first fragment).
+ * frag: The packet is fragmented.
+ * proto: L4 protocol (the last nexthdr for IPv6).
+ * fragment id: 32-bit for IPv6, 16-bit for IPv4 (lower 16 bits).
```

Comment:
Q: In the current encoding, IPv6 fraginfo and IPv4 fraginfo with same ID are indistinguishable because it doesn't encode IP version. Seems like we only use in the IP version-specific paths, so maybe ok?